migrations/1692891772000_brc20-events-types.ts.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { MigrationBuilder, ColumnDefinitions } from 'node-pg-migrate';
+
+export const shorthands: ColumnDefinitions | undefined = undefined;
+
+export function up(pgm: MigrationBuilder): void {
+  // 1) Add the new columns to the table
+  pgm.addColumns('brc20_events', {
+    genesis_location_id: {
+      type: 'bigint',
+      references: '"locations"',
+      onDelete: 'CASCADE',
+      notNull: false,
+      unique: true, // only one event exists per location
+    },
+    operation: {
+      type: 'text', // enum-style: ['deploy', 'mint', 'prepare_transfer', 'transfer']
+      notNull: false,
+    },
+  });
+
+  // 2) Make sure all events exist in the table
```

Comment:
I think we could do away with these migration queries because we'll be starting a new index from scratch and I don't think they'll cover all cases (see below).